frappe/desk/form/load.py

```diff
@@ -94,30 +94,78 @@ def get_docinfo(doc=None, doctype=None, name=None):
 	automated_messages = filter(lambda x: x['communication_type'] == 'Automated Message', all_communications)
 	communications_except_auto_messages = filter(lambda x: x['communication_type'] != 'Automated Message', all_communications)
 
-	frappe.response["docinfo"] = {
+	docinfo = frappe._dict(user_info = {})
+
+	add_comments(doc, docinfo)
+
+	docinfo.update({
 		"attachments": get_attachments(doc.doctype, doc.name),
-		"attachment_logs": get_comments(doc.doctype, doc.name, 'attachment'),
 		"communications": communications_except_auto_messages,
 		"automated_messages": automated_messages,
-		'comments': get_comments(doc.doctype, doc.name),
 		'total_comments': len(json.loads(doc.get('_comments') or '[]')),
 		'versions': get_versions(doc),
 		"assignments": get_assignments(doc.doctype, doc.name),
-		"assignment_logs": get_comments(doc.doctype, doc.name, 'assignment'),
 		"permissions": get_doc_permissions(doc),
 		"shared": frappe.share.get_users(doc.doctype, doc.name),
-		"info_logs": get_comments(doc.doctype, doc.name, comment_type=['Info', 'Edit', 'Label']),
-		"share_logs": get_comments(doc.doctype, doc.name, 'share'),
-		"like_logs": get_comments(doc.doctype, doc.name, 'Like'),
-		"workflow_logs": get_comments(doc.doctype, doc.name, comment_type="Workflow"),
 		"views": get_view_logs(doc.doctype, doc.name),
 		"energy_point_logs": get_point_logs(doc.doctype, doc.name),
 		"additional_timeline_content": get_additional_timeline_content(doc.doctype, doc.name),
 		"milestones": get_milestones(doc.doctype, doc.name),
 		"is_document_followed": is_document_followed(doc.doctype, doc.name, frappe.session.user),
 		"tags": get_tags(doc.doctype, doc.name),
-		"document_email": get_document_email(doc.doctype, doc.name)
-	}
+		"document_email": get_document_email(doc.doctype, doc.name),
+	})
+
+	update_user_info(docinfo)
+
+	frappe.response["docinfo"] = docinfo
+
+def add_comments(doc, docinfo):
```

Comment:
cleaned up multiple queries into a single one (for each comment type, there was a separate query)